numpy/lib/function_base.py

```diff
@@ -3870,15 +3870,17 @@ def _quantile_is_valid(q):
     return True
 
 
+def _lerp(a, b, t, out=None):
+    """ Linearly interpolate from a to b by a factor of t """
+    return add(a*(1 - t), b*t, out=out)
+
+
 def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
                            interpolation='linear', keepdims=False):
     a = asarray(a)
-    if q.ndim == 0:
-        # Do not allow 0-d arrays because following code fails for scalar
-        zerod = True
-        q = q[None]
-    else:
-        zerod = False
+
+    # Helper to undo any decay that the ufunc performed (see gh-13105). This
```

Comment:
Some comment lost. can you make it "decay to scalar", the word feels not very clear on its own.

A bit funny and for now maybe it does not matter, but I think in the long run, we may need to make this type of thing `np.asanyarray(..., dtype=a.dtype)`

Reply:
In the long run we need `np.leave_wrapped` or similar, in my opinion :)

Reply:
I've expanded the comment considerably, let me know what you think